checkbox-support/checkbox_support/tests/test_psnr.py

```diff
@@ -0,0 +1,186 @@
+
+    @patch("checkbox_support.scripts.psnr.argparse.ArgumentParser.parse_args")
+    def test_psnr_args_with_custom_args(self, mock_parse_args):
+        mock_parse_args.return_value = Namespace(
+            reference_file="ref.jpg",
+            test_file="test.jpg",
+            show_psnr_each_frame=True,
+        )
+
+        args = psnr_args().parse_args()
+
+        self.assertEqual(args.reference_file, "ref.jpg")
+        self.assertEqual(args.test_file, "test.jpg")
+        self.assertTrue(args.show_psnr_each_frame)
+
+
+class TestGetFrameResolution(unittest.TestCase):
+    @patch("checkbox_support.scripts.psnr.cv2.VideoCapture")
+    def test_get_frame_resolution(self, mock_videocapture):
+        mock_capture = mock_videocapture.return_value
+        mock_capture.get.side_effect = [100, 200]
+        width, height = _get_frame_resolution(mock_capture)
+        self.assertEqual(width, 100)
+        self.assertEqual(height, 200)
+
+
+class TestGetPSNR(unittest.TestCase):
+
+    def create_image(self, width, height, color):
+        """Creates an image with the specified color."""
+        return np.full((height, width, 3), color, dtype=np.uint8)
+
+    def test_identical_images(self):
+        img1 = self.create_image(100, 100, 255)
+        img2 = self.create_image(100, 100, 255)
+        self.assertEqual(_get_psnr(img1, img2), 0.0)
+
+    def test_different_images(self):
+        img1 = self.create_image(100, 100, 255)
+        img2 = self.create_image(100, 100, 0)
+        self.assertNotEqual(_get_psnr(img1, img2), 0.0)
+
+
+class TestGetAveragePSNR(unittest.TestCase):
+    def test_get_average_psnr_file_not_found(self):
+        with patch(
+            "checkbox_support.scripts.psnr.cv2.VideoCapture"
+        ) as mock_videocapture:
+            mock_capture = mock_videocapture.return_value
+            mock_capture.isOpened.return_value = False
+            with self.assertRaises(SystemExit):
+                get_average_psnr("nonexistent_file.mp4", "test_file.mp4")
+
+    @patch("checkbox_support.scripts.psnr.cv2.VideoCapture")
+    def test_zero_frames(self, mock_VideoCapture):
+        mock_video_capture = MagicMock()
+        mock_VideoCapture.return_value = mock_video_capture
+        mock_video_capture.isOpened.return_value = True
+        mock_video_capture.get.return_value = 0  # Zero frames in the video
+
+        with self.assertRaises(SystemExit):
+            get_average_psnr("ref.mp4", "test.mp4")
+
+    @patch("checkbox_support.scripts.psnr._get_frame_resolution")
+    @patch("checkbox_support.scripts.psnr.cv2.VideoCapture")
+    def test_get_average_psnr_different_dimensions(
+        self, mock_videocapture, mock_get_frame_resolution
+    ):
+        mock_capture_ref = mock_videocapture.return_value
+        mock_capture_test = mock_videocapture.return_value
+        mock_capture_test.isOpened.return_value = True
+        mock_capture_ref.isOpened.return_value = True
```

Comment:
These two are the same mock, you could use just one:
```suggestion
    def test_get_average_psnr_different_dimensions(
        self, mock_vc, mock_get_frame_resolution
    ):
        mock_vc.return_value.isOpened.return_value = True
```